packages/devextreme/js/__internal/ui/overlay/m_overlay.ts

```diff
@@ -1318,7 +1463,7 @@ class Overlay<
         this._toggleHideTopOverlayCallback(false);
         // @ts-expect-error ts-error
         this._initHideTopOverlayHandler(value);
-        this._toggleHideTopOverlayCallback(this.option('visible'));
+        this._toggleHideTopOverlayCallback(Boolean(this.option('visible')));
```

Comment:
and allow _toggleHideTopOverlayCallback to receive boolean | undefined argument
```suggestion
        const { visible } = this.option();
        this._toggleHideTopOverlayCallback(visible))
```

Reply:
eca8905e08c0af1d0eedada4c1e167e77335af2c, 54fb3761139b89233aabdcf60cfc00701960f9b6, 331517e56476f40cbf591e671dcd6b8e43edb8ca, 38d39427a4658cbf6852f4fe72f979b6c9c54a49